nullaway/src/test/java/com/uber/nullaway/NullAwayAndroidTest.java

```diff
@@ -0,0 +1,211 @@
+package com.uber.nullaway;
+
+import com.google.errorprone.CompilationTestHelper;
+import com.sun.tools.javac.main.Main;
+import java.util.Arrays;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link com.uber.nullaway.NullAway}. */
+@RunWith(JUnit4.class)
+@SuppressWarnings("CheckTestExtendsBaseClass")
+public class NullAwayAndroidTest {
+  @Rule public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private CompilationTestHelper compilationHelper;
+
+  @Before
+  public void setup() {
+    compilationHelper = CompilationTestHelper.newInstance(NullAway.class, getClass());
+    compilationHelper.setArgs(
+        Arrays.asList(
+            "-d",
+            temporaryFolder.getRoot().getAbsolutePath(),
+            "-XepOpt:NullAway:KnownInitializers="
+                + "com.uber.nullaway.testdata.CheckFieldInitNegativeCases.Super.doInit,"
+                + "com.uber.nullaway.testdata.CheckFieldInitNegativeCases"
+                + ".SuperInterface.doInit2",
+            "-XepOpt:NullAway:AnnotatedPackages=com.uber,com.ubercab,io.reactivex",
+            // We give the following in Regexp format to test that support
+            "-XepOpt:NullAway:UnannotatedSubPackages=com.uber.nullaway.[a-zA-Z0-9.]+.unannotated",
+            "-XepOpt:NullAway:ExcludedClasses="
+                + "com.uber.nullaway.testdata.Shape_Stuff,"
+                + "com.uber.nullaway.testdata.excluded",
+            "-XepOpt:NullAway:ExcludedClassAnnotations=com.uber.nullaway.testdata.TestAnnot",
+            "-XepOpt:NullAway:CastToNonNullMethod=com.uber.nullaway.testdata.Util.castToNonNull",
+            "-XepOpt:NullAway:ExternalInitAnnotations=com.uber.ExternalInit",
+            "-XepOpt:NullAway:ExcludedFieldAnnotations=com.uber.ExternalFieldInit"));
+  }
+
+  // Core Fragment tests
+
+  @Test
+  public void coreFragmentSuccess() {
+    initialiseAndroidCoreClasses();
+    compilationHelper
+        .addSourceFile("androidstubs/core/Fragment.java")
+        .addSourceFile("android-success/CoreFragment.java")
+        .doTest();
+  }
+
+  @Test(expected = AssertionError.class)
+  public void coreFragmentMissingOnAttachError() {
+    initialiseAndroidCoreClasses();
+    compilationHelper
+        .addSourceFile("androidstubs/core/Fragment.java")
+        .addSourceFile("android-error/CoreFragmentWithoutOnAttach.java")
```

Comment:
If you want to catch the expected error here, the right way is to add a comment to `CoreFragmentWithoutOnAttach` just before the line with the NullAway error, with the following form:

`// BUG: Diagnostic contains: error message`

See e.g., `NullAwayPositiveCases.java` which has a lot of these comments

Reply:
Great! This is exactly what I wanted :-) 

Thanks for the help